pkgs/development/compilers/graalvm/community-edition/mkGraal.nix

```
@@ -33,16 +32,19 @@
 , cairo
 , glib
 , gtk3
+, writeShellScript
+, jq
+, gnused
 }:
 
 assert useMusl -> stdenv.isLinux;
 
 let
-  platform = {
-    aarch64-linux = "linux-aarch64";
-    x86_64-linux = "linux-amd64";
-    x86_64-darwin = "darwin-amd64";
-  }.${stdenv.system} or (throw "Unsupported system: ${stdenv.system}");
+  platform = config.${stdenv.hostPlatform.system} or (throw "Unsupported system: ${stdenv.hostPlatform.system}");
+  version = platform.version;
+  name = "graalvm${javaVersion}-ce";
+  sourcesFilename = "${name}-sources.json";
+  sources = builtins.fromJSON (builtins.readFile (./. + "/${sourcesFilename}"));
```

This should also work:
```suggestion
  sources = builtins.fromJSON (builtins.readFile (./${sourcesFilename}));
```

---

Thanks, I'll change it.

---

Looks like this change fails the output comparison test. Is it ok if I revert?

---

Sure.